hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -331,27 +334,28 @@ boolean needsBalance(TableName tableName, BalancerClusterState cluster) {
         LOG.trace("{} not needed", c.getClass().getSimpleName());
         continue;
       }
-      sumMultiplier += multiplier;
-      total += c.cost() * multiplier;
+      if (cost < minCostNeedBalance)
+      {
+        LOG.debug("Imbalance of {} on the scale of [0, 1] is {} < threshold ({}).",
+          c.getClass().getSimpleName(), cost, minCostNeedBalance);
+      }
```

Comment:
What's the goal here? It looks like the trace level output of the `functionCost()` call has been removed in favor of debug output here? Will need to include the weights and output for cost functions that were over the threshold otherwise there'll be no way to reproduce why the balancer decided to skip from just the logs.